Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,658 @@
+    {
+        htmlElement.innerText = result;
+        return true;
+    }
+    return false;
+}
+
+/**
+* send method to invoke on LVA module
+* @param {jsonObject} fullPayload - payload to be sent in request to server
+* @returns {Promise<any>}
+*/
+function invokeLVAMethod(fullPayload)
+{
+    return new Promise((resolve, reject) =>
+    {
+        let request = prepareRequest(`http://localhost:${PORT}/runMethod`);
+        // event listener. When request readyState changes, place response in the output box. If response is done (readyState=4) then update global vars accordingly
+        request.onreadystatechange = function ()
+        {
+            //on successful response. Result is object like [{methodName: 'GraphTopologyList'}, {value: 'long JSON object....'}]
+            if (request.readyState == 4)
+            {
+                if (request.status == 200)
+                {
+                    var methodName = JSON.parse(request.response)[0].method;
+                    var result = JSON.stringify(JSON.parse(request.response)[1]);
+                    var isFromOutputPage = displayMethodOutput(result);
+                    if (JSON.parse(result).status < 250)
+                    {
+                        switch (methodName)
+                        {
+                            case 'GraphTopologyDelete':
+                                delete graphTopologies[fullPayload.Payload.name];
+                                break;
+                            case 'GraphInstanceDelete':
+                                delete graphInstances[fullPayload.Payload.name];
+                                break;
+                            case 'GraphInstanceList':
+                                updateGraphsandInstances(methodName, JSON.parse(request.response)[1]);
+                                break;
+                            case 'GraphTopologyList':
+                                updateGraphsandInstances(methodName, JSON.parse(request.response)[1]);
+                                break;
+                            case 'GraphInstanceSet':
+                                graphInstances[fullPayload.Payload.name] = fullPayload;
+                                break;
+                            case 'GraphTopologySet':
+                                // only alert and run display media graphs if request is coming from mediagraphs.html page
+                                graphTopologies[fullPayload.Payload.name] = fullPayload;
+                                if (!isFromOutputPage)
+                                {
+                                    alert("Success!");
+                                    displayMediaGraphs();
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                        resolve(result);
+                    }
+                }
+                else
```

Comment:
looks like right now if we get status 4 and a non-200 code we do nothing. the only handling is for status not 4, which i think we don't want anyways (none of the other methods have it).